src/main/scala/nl.knaw.dans.easy.deposit/authentication/ServletEnhancedLogging.scala

```diff
@@ -15,42 +15,75 @@
+    // saved at https://web.archive.org/web/20181030102418/
+    val maskedRemoteAddr = request.getRemoteAddr.replaceAll("[0-9]+[.][0-9]+$","**.**")
+    logger.info(s"${ request.getMethod } ${ request.getRequestURL } remote=$maskedRemoteAddr; params=$maskedParams; headers=$headers")
   }
 }
 object ServletEnhancedLogging extends DebugEnhancedLogging {
 
-  implicit class RichActionResult(actionResult: ActionResult)(implicit request: HttpServletRequest) {
-    def logResponse: ActionResult = logResult(actionResult)
+  implicit class RichActionResult(actionResult: ActionResult)(implicit request: HttpServletRequest, response: HttpServletResponse) {
+    /**
+     * @example halt(BadRequest().logResponse)
+     * @example Ok().logResponse
+     * @return this
+     */
+    def logResponse: ActionResult = {
+      // Disadvantage of this method: developers might forget to call,
+      // but an after method of the trait is not executed after a halt anyway.
+      // Halt is typically used during authentication.
+      // Another disadvantage of an after method: it might not have the response to log,
+      // which should have been copied into "implicit response: HttpServletResponse".
+      // See the last extensive readme version (documentation moved into an incomplete book and guides)
+      // https://github.com/scalatra/scalatra/blob/6a614d17c38d19826467adcabf1dc746e3192dfc/README.markdown
+      // sections #filters #action
+      val authHeaders = response.getHeaderNames.toArray().map{
+        case "Expires" => "Expires " + response.getHeader("Expires")
+        case headerName => headerName
+      }
+      logger.info(s"${ request.getMethod } returned status=${ actionResult.status }; authHeaders=$authHeaders; actionHeaders=${ actionResult.headers }")
+      actionResult
+    }
   }
 
-  implicit class RichTriedActionResult(tried: Try[ActionResult])(implicit request: HttpServletRequest) {
-    // TODO to preserve actionResult into and beyond after filters, copy it into "implicit response: HttpServletResponse"
-    // See the last extensive readme version (documentation moved into an incomplete book and guides)
-    // https://github.com/scalatra/scalatra/blob/6a614d17c38d19826467adcabf1dc746e3192dfc/README.markdown
-    // sections #filters #action
+  implicit class RichTriedActionResult(tried: Try[ActionResult])(implicit request: HttpServletRequest, response: HttpServletResponse) {
+    // convenience replacing: .getOrRecover(???).logResponse
+    // what happened to be the standard pattern within this project
     def getOrRecoverResponse(recover: Throwable => ActionResult): ActionResult = {
-      // the signature is more specific than in nl.knaw.dans.lib.error and comes with the trait, not with just an import
-      logResult(tried match {
+      // the signature is more specific than for nl.knaw.dans.lib.error.TryExtensions.getOrRecover
+      // it comes with the trait, not with just an import
+      (tried match {
```

Comment:
I considered to drop this one. Opinions please.

Reply:
You mean the change from `logResult(...)` to `xxx.logResponse`? I mean, they're kind of the same, right. The only difference is that it requires more wrapping in implicit classes in the second case, so this might be slightly more expensive. But that is very, very minor. I wouldn't worry about that.

Reply:
I mean to drop the complete implicit class. That means in the servlets I should replace `getORecoverResponse(???)` with `getOrRecover(???).logResponse`

Reply:
👍 great idea! That would mean we just have one method for logging the response: `.logResponse`.